consensus/scp/src/node.rs

```diff
@@ -15,28 +15,25 @@ use mc_crypto_digestible::Digestible;
 use sha3::Sha3_256;
 use std::{collections::BTreeSet, fmt::Display, time::Duration};
 
-/// Max number of pending slots to store.
-const MAX_PENDING_SLOTS: usize = 10;
-
 /// Max number of externalized slots to store.
-const MAX_EXTERNALIZED_SLOTS: usize = 10;
+pub const MAX_EXTERNALIZED_SLOTS: usize = 10;
 
 /// Number of last seen messages to keep track of.
 const LAST_SEEN_HISTORY_SIZE: usize = 1000;
 
 /// A node participates in federated voting.
-pub struct Node<V: Value, ValidationError: Display> {
+pub struct Node<V: Value, ValidationError: Clone + Display> {
     /// Local node ID.
     pub ID: NodeID,
 
     /// Local node quorum set.
     pub Q: QuorumSet,
 
-    /// Map of last few slot indexes -> slots.
-    pub pending: LruCache<SlotIndex, Slot<V, ValidationError>>,
+    /// The current slot that this node is attempting to reach consensus on.
+    pub current_slot: Slot<V, ValidationError>,
 
-    /// Map of last few slot indexes -> externalized slots.
-    pub externalized: LruCache<SlotIndex, ExternalizePayload<V>>,
+    /// Previous, externalized slots, ordered by increasing slot index.
+    pub externalized_slots: Vec<Slot<V, ValidationError>>,
```

Comment:
I think I agree that there's no advantage to using lrucache here, since nodes must process all slots in sequence if they fall behind. Does this mean we can drop it as a requirement in Cargo.toml for the crate?